corehq/apps/users/views/mobile/users.py

```diff
@@ -1497,18 +1505,41 @@ def user_id(self):
     @memoized
     def form(self):
         if self.request.method == 'POST':
-            return MobileWorkerAccountConfirmationForm(self.request.POST)
+            return MobileWorkerAccountConfirmationBySMSForm(self.request.POST)
         else:
-            if not self.is_invite_valid():
-                raise ValidationError("Invite has expired.")
-            return MobileWorkerAccountConfirmationForm(initial={
+            return MobileWorkerAccountConfirmationBySMSForm(initial={
                 'username': self.user.raw_username,
                 'full_name': self.user.full_name,
-                'email': self.user.email,
+                'email': "",
```

Comment:
Don't you want to preserve any existing email addresses?

If you intend to remove existing email addresses then instead of hiding the form field you could just drop it ...
```python
class MobileWorkerAccountConfirmationBySMSForm(BaseUserInvitationForm):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.fields.pop('email', None)
```
... but I'm not sure why you would want to delete mobile workers' email addresses.

Reply:
This will be used to populate the fields when a *new* user lands on signup page using the invite link. Also, user account will be created by the project administrator with no email address. Usecase here assumes that the mobile workers may not have an email address.